src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -96,63 +96,23 @@ public SonarWebService(IDownloader downloader, string server, ILogger logger)
         }
 
         /// <summary>
-        /// Retrieves rule keys of rules having a given language and that are not activated in a given profile.
-        ///
-        /// </summary>
-        /// <param name="qprofile">Quality profile key.</param>
-        /// <param name="language">Rule Language.</param>
-        /// <returns>Non-activated rule keys, including repo. Example: csharpsquid:S1100</returns>
-        public async Task<IList<SonarRule>> GetInactiveRules(string qprofile, string language)
-        {
-            var fetched = 0;
-            var total = 0;
-            var ruleList = new List<SonarRule>();
-
-
-            for (int page = 1; page <= 20; page++)
-            {
-                var ws = GetUrl("/api/rules/search?f=repo,name,severity,lang,internalKey,templateKey,params&ps=500&activation=false&qprofile={0}&p={1}&languages={2}", qprofile, page.ToString(), language);
-                this.logger.LogDebug(Resources.MSG_FetchingInactiveRules, qprofile, language, ws);
-
-                ruleList.AddRange(await DoLogExceptions(async () =>
-                {
-                    var contents = await this.downloader.Download(ws);
-                    var json = JObject.Parse(contents);
-                    total = json["total"].ToObject<int>();
-                    fetched += json["ps"].ToObject<int>();
-                    var rules = json["rules"].Children<JObject>();
-
-                    return rules.Select(r => new SonarRule(r["repo"].ToString(), ParseRuleKey(r["key"].ToString()), false));
-                }, ws));
-
-                if (fetched >= total)
-                {
-                    break;
-                }
-
-            }
-
-            return ruleList;
-        }
-
-        /// <summary>
-        /// Retrieves active rules from the quality profile with the given ID, including their parameters and template keys.
-        ///
+        /// Retrieves rules from the quality profile with the given ID, including their parameters and template keys.
         /// </summary>
-        /// <param name="qprofile">Quality profile id.</param>
+        /// <param name="qProfile">Quality profile id.</param>
         /// <returns>List of active rules</returns>
-        public async Task<IList<SonarRule>> GetActiveRules(string qprofile)
+        public async Task<IList<SonarRule>> GetRules(string qProfile)
```

Comment:
Instead of `qProfile` I would chose `profileId` considering it's string one could get confused and provide the profile name instead.

Reply:
It's aligned with SC/SQ API `&qprofile=` and documented. I just moved the documentation to the interface where it should be for users 🙂